### PR TITLE
gdb: adds support for Guile.

### PR DESCRIPTION
**Pull request by mkohlhaas**
Source highlighting is not included as it looks really crappy.

https://github.com/void-linux/void-packages/issues/23051

### CODE REVIEW

File: srcpkgs/gdb/template [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 pycompile_dirs="/usr/share/gdb/python"
-configure_args="--disable-werror --disable-nls --with-system-readline
+configure_args="--disable-werror --disable-nls --with-system-readline --with-guile=guile-2.0
  --with-system-gdbinit=/etc/gdb/gdbinit --with-system-zlib $(vopt_enable gdbserver)
  $(vopt_if static 'CFLAGS=-static CXXFLAGS=-static LDFLAGS=-static')
  $(vopt_if python --with-python=/usr/bin/python3)"
-hostmakedepends="texinfo perl $(vopt_if python python3-devel)"
-makedepends="ncurses-devel zlib-devel readline-devel expat-devel
+hostmakedepends="texinfo perl $(vopt_if python python3-devel) guile" 
+makedepends="ncurses-devel zlib-devel readline-devel expat-devel guile-devel pkg-config
```
Comment (abenson) [inline srcpkgs/gdb/template:12]: `pkg-config` should be in hostmakedepends

**Comment by abenson** [comment]
Please amend your commit message to following our [conventions](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes)

**Comment by abenson** [comment]
Why'd you close the PR?

**Comment by mkohlhaas** [comment]
Guile repl is crashing gdb.

**Comment by abenson** [comment]
Keep it open, find a fix, push the fix.

**Pull request closed without merging**